fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexQueryPlan.java

```diff
@@ -107,8 +107,19 @@ public static IndexKeyValueToPartialRecord getToPartialRecord(@Nonnull Index ind
                     .addLogInfo(LogMessageKeys.SCAN_TYPE, scanType);
         }
 
-        builder.addRegularCopier(new LuceneIndexKeyValueToPartialRecordUtils.LuceneAutoCompleteCopier());
+        builder.addRegularCopier(new LuceneIndexKeyValueToPartialRecordUtils.LuceneAutoCompleteCopier(scanType.equals(LuceneScanTypes.BY_LUCENE_AUTO_COMPLETE),
+                recordType.getPrimaryKey()));
 
         return builder.build();
     }
+
+    /**
+     * Auto-Complete and Spell-Check scan has their own implementation for {@link IndexKeyValueToPartialRecord} to build partial records,
+     * so they are not appropriate for the optimization by {@link com.apple.foundationdb.record.query.plan.plans.RecordQueryCoveringIndexPlan}.
```

Comment:
I believe the consequence of this design is that auto-complete queries will now get partial records that have:

1. Any grouping keys
2. The primary key
3. The completed text

And nothing else. This means that if the user wants other fields from the record, they have to go look up the record by primer key, I suppose. That...may be the best we can do here, in that there's sort of an amibiguity baked into the way that partial records are generated, where the auto-completed text can't be returned in the same record as other text fields but in a way that still allows the auto-completed text to be distinguished. (In SQL, say, this would probably be some pseudo-column added to the result set that was labeled as the result of the function evaluation, I think.)

At some points we should definitely document what the contract is here so that downstream users can reason about this special kind of scan, but this is probably fine for now.

Reply:
Added java doc for the method ```executePlan```.